designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/UIProcessObjectsFactory.scala

```diff
@@ -51,10 +51,12 @@ object UIProcessObjectsFactory {
 
     val dynamicComponentsConfig = uiProcessDefinition.allDefinitions.mapValuesNow(_.componentConfig)
 
+    val additionalComponentsUIConfig = modelDataForType.additionalComponentsUIConfigProvider.getAllForCategory(processingType)
+
     val fragmentsComponentsConfig = fragmentInputs.mapValuesNow(_.objectDefinition.componentConfig)
     //we append fixedComponentsConfig, because configuration of default components (filters, switches) etc. will not be present in dynamicComponentsConfig...
     //maybe we can put them also in uiProcessDefinition.allDefinitions?
-    val finalComponentsConfig = ComponentDefinitionPreparer.combineComponentsConfig(fragmentsComponentsConfig, fixedComponentsUiConfig, dynamicComponentsConfig)
+    val finalComponentsConfig = ComponentDefinitionPreparer.combineComponentsConfig(fragmentsComponentsConfig, additionalComponentsUIConfig, fixedComponentsUiConfig, dynamicComponentsConfig) // TODO consider the ordering
```

Comment:
NOTE:
this goes into `UIProcessObjects` as `componentsConfig`, but I'm not sure if FE relies on that field for what it displays in the parameters' form

I count at least 4 sources of parameter information in the `UIProcessObjects` alone:
1. componentGroups -> *group* -> components -> *name* -> node -> service/ref/?-> parameters
2. processDefinition -> *group* -> *name* -> parameters
3. processDefinition -> *group* -> *name* -> componentConfig -> params
4. componentsConfig -> *name* -> params

This draft uses the 4th option, as it was most convenient in BE code.